storage_engine/page_store/src/lib.rs

```diff
@@ -0,0 +1,698 @@
+// pub mod file_commander;
+pub mod page_usize;
+
+use util::error::{Error, ErrorType};
+use util::{Env, ENV};
+
+use page_usize::{page_usize_from_le_bytes, page_usize_to_le_bytes, PageUsizeType};
+
+pub struct PageStoreHeaders {
+    page_len: u64, // 8 bytes to store the page length in little endian
+}
+impl PageStoreHeaders {
+    pub fn new(page_len: usize) -> Self {
+        PageStoreHeaders {
+            page_len: page_len as u64,
+        }
+    }
+    pub fn to_bytes(&self) -> [u8; 8] {
+        u64::to_le_bytes(self.page_len)
+    }
+    pub fn from_bytes(bytes: [u8; 8]) -> Self {
+        let page_len = u64::from_le_bytes(bytes);
+        PageStoreHeaders { page_len }
+    }
+}
+
+const PAGE_HEADER_LEN: usize = std::mem::size_of::<PageStoreHeaders>();
+
+pub struct PageSettings {
+    page_size_type: PageUsizeType,
+    page_capacity: usize,
+    page_len: usize,
+}
+impl PageSettings {
+    pub fn new(page_len: usize) -> Self {
+        let page_size_type = PageUsizeType::for_max_usize(page_len);
+        let page_capacity = page_len - PageUsizeType::size_of(page_size_type);
+        PageSettings {
+            page_size_type,
+            page_capacity,
+            page_len,
+        }
+    }
+}
+
+pub struct PageStore {
+    page_settings: PageSettings,
+    /// last_page_index + 1 is the number of pages
```

Comment:
'number of pages' means page_count?

Reply:
Oops.

So page_count indicates total number of pages ever allocated in the storage.